types/carbon-components-react/lib/components/CodeSnippet/CodeSnippet.d.ts

```diff
@@ -28,7 +28,7 @@ export interface CodeSnippetMultiProps extends SharedProps, Omit<ReactDivAttr, '
     type: 'multi';
 }
 
-export interface CodeSnippetInlineProps extends SharedProps, Omit<CopyProps, 'children' | 'type'> {
+export interface CodeSnippetInlineProps extends SharedProps, Omit<CopyProps, 'type' | keyof SharedProps> {
```

Comment:
The explicit `children` removal should probably stay in case it gets removed from `SharedProps`, no?